src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -1213,11 +1213,13 @@ private static bool HadLambdaConversionError(BindingDiagnosticBag diagnostics, A
                         new FormattedSymbol(UnwrapIfParamsArray(parameter, isLastParameter), SymbolDisplayFormat.CSharpErrorMessageNoParameterNamesFormat));
                 }
             }
-            else if (refArg != refParameter && !(refArg == RefKind.None && refParameter == RefKind.In))
+            else if (refArg != refParameter &&
+                !(refArg == RefKind.None && refParameter == RefKind.In) &&
```

Comment:
>(refArg == RefKind.None && refParameter == RefKind.In) [](http://example.com/codeflow?start=17&length=54)

It looks like scenario when ```(refArg == RefKind.Ref && refParameter == RefKind.In)``` is a valid scenario now, depending on the language version. Are we testing this code path with scenario like that for different versions?

Reply:
Good catch. Will extend test `PassingArgumentsToInParameters_RefKind_None_WrongType` and fix the logic.